### PR TITLE
Display persistent data inside Loot Tracker UI

### PR DESCRIPTION
**Pull request by TheStonedTurtle**
Record quantity is limited to the default return value of the API endpoint

### CODE REVIEW

**Comment by TheStonedTurtle** [comment]
#7071 fixed the API data issue

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 1]
```diff
@@ -32,10 +32,33 @@
 @ConfigGroup("loottracker")
 public interface LootTrackerConfig extends Config
 {
+	@ConfigItem(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: I dont think this needs a toggle?

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java [hunk 2]
```diff
@@ -32,10 +32,33 @@
...
+		name = "Persistent data",
+		description = "Configures whether loot tracker data should persist between sessions.<br/> Must be Logged in.",
+		position = 0
+	)
+	default boolean persistentData()
+	{
+		return false;
+	}
+
+	@ConfigItem(
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: This should be enabled by default, and it should mention that it requires login in name (e.g Historical data (Requires login!)

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java [hunk 3]
```diff
@@ -209,6 +213,34 @@ protected void startUp() throws Exception
...
+				{
+					switch (client.getGameState())
+					{
+						case STARTING:
+						case UNKNOWN:
+							return false;
+					}
+
+					// Run blocking web requests on new thread
+					new Thread(() ->
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java]: This should use executor service, and the entire thing should be limited to maybe 100-1000 loot records

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java (hunk 1 above)
Comment (TheStonedTurtle) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: Some people may not want to persist data between sessions so i figured why store their data for no reason? I can revert this config change if you'd like.

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java (hunk 3 above)
Comment (TheStonedTurtle) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java]: The get API route limit defaults to `1024` already so should be fine as is.

I've also updated the `new Thread()` to use `ScheduledExecutorService.submit()` instead

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java (hunk 1 above)
Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: Well, we don't have toggle in master, so why add it now. And for people that doont want to see the persisted data, they will just disable the other toggle

Comment (TheStonedTurtle) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerConfig.java]: The changes in master were meant to just test the API performance so we didn't want to announce it when we were still testing the feasibility.

Either way I've removed that config option as requested.

File: runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java [hunk 4]
```diff
@@ -392,4 +427,31 @@ boolean isIgnored(String name)
...
+
+	private Collection<LootTrackerRecord> convertToLootTrackerRecord(final Collection<LootRecord> records)
+	{
+		Collection<LootTrackerRecord> trackerRecords = new ArrayList<>();
+		for (LootRecord record : records)
+		{
+			// Convert GameItem drops into LootTrackerItems
+			LootTrackerItem[] drops = record.getDrops().stream().map(itemStack ->
+			{
+				final ItemComposition itemComposition = itemManager.getItemComposition(itemStack.getId());
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java]: Why is all of this logic duplicated from above?

Comment (TheStonedTurtle) [reply runelite-client/src/main/java/net/runelite/client/plugins/loottracker/LootTrackerPlugin.java]: No idea, should have separated it into a single function that is used in both places. I'll do so now unless you are working on this.

**Pull request merged**